packages/as-proba/assembly/probability/sampler.ts

```diff
@@ -1,32 +1,46 @@
 import { randomInt } from './random';
 
 /**
- * Generates samples based on a probability distribution.
+ * This module exports a class named Sampler that allows the generation of observations based on a
+ * probability distribution.
  *
- * This class shall be extended with your own probability.
+ * @remarks
+ * - This class shall be extended with your own probability.
+ * - Two methods are available to generate observations: {@link rejectionSampling} and
+ * {@link inverseCumulativeDistributionSampling}.
+ *    These methods use different techniques to generate observations and require the implementation
+ *    of a probability function.
  */
 export class Sampler {
   _bounderies: Float64Array;
 
   /**
-   * Instanciates a sampler.
+   * Creates an instance of the Sampler class.
    *
-   * This constructor calls seedRandom.
+   * @remarks
+   * - The constructor uses an optional seed for the Math.random() function.
+   * - The _bounderies field is initialized with an empty Float64Array.
+   *
+   * @param s - seed for random Math.random function. Default value is 0.
    *
-   * @param s - seed for random Math.random function.
    */
   constructor(s: u64 = 0) {
     Math.seedRandom(s);
     this._bounderies = new Float64Array(0);
   }
 
   /**
-   * Returns the probability of given sample.
+   * Returns the probability of the given sample.
+   *
    *
-   * Probability function doesn't need to be normalized,
-   * but the greatest probability of the distribution must be knowned.
+   * @remarks
+   * - This method should be overridden in the child class to match the desired probability distribution.
```

Comment:
maybe a @link to binomial exemple ?

Reply:
Not necessarily, the dev chooses the distribution he wants

Reply:
need confirmation

Reply:
Has anything changed in the meantime ? Could you clarify the problem ?

Reply:
This documentation here is meant to guide the user on how to implement a 'Sampler' extended class, so I proposed to add a link to the 'Binomial' implementation for the users.
However It might be better to put such link in the interface description itself.
That's the discussion @Ben-Rey, do you agree with me ?

Reply:
There is an example of using sampler in sampler.spec.ts.
It's a good idea to link an example to the tests or the binomial class that extends sampler. Just make it clear that this is just an example of using it.